app/ExposureNotificationContext.tsx

```diff
@@ -1,15 +1,16 @@
-import React, { createContext, useState } from 'react';
+import React, { createContext, useState, useEffect } from 'react';
 
+import { isGPS } from './COVIDSafePathsConfig';
 import { BTNativeModule } from './bt';
 
-export type ENAuthorizationStatus = 'authorized' | 'notAuthorized';
+export type ENAuthorizationStatus = `NOT_AUTHORIZED` | `AUTHORIZED`;
```

Comment:
Sure - and now that you mention it, this should be `EnabledStatus` (since `AuthorizationStatus` belongs to the OS)

Reply:
`export type EnabledStatus = `DISABLED` | `ENABLED`;`

Reply:
maybe even: `export type Status = DISABLED | ENABLED` but that might not be descriptive enough.

`export type Authorization = DISABLED | ENABLED`

is another one that might be good.

Reply:
Ah, I see the disconnect here. this is the `ExposureNotifications` Context so i see it as EN only. ideally wed never have any gps stuff here. OR are you thinking that this component will eventually become something generic over the tracing strategy?  if so should we change the name of the component?

Reply:
ah - `EN` needs to be both "authorized" for the OS and "enabled" for the app; it's possible for it to be `authorized` but not `enabled` for the app. The user will see the confirmation dialog only if `EN` have not yet been "authorized" for the OS. From then on, the user must enable/disable `EN` for specific apps from the settings detail for the specific app. They can also "authorize"/"unauthorize" `EN` from global settings

Reply:
got it, so we have the 4 possible states as mentioned above.  As far as the type we want here, i could see it going either way depending on how we want to present this info to the user. If we want a specific message for each of the 4 cases it might make sense to have a type along the lines of

```typescript
type Authorization = Authorized | Unauthorized
type Enablement = Enabled | Disabled

type ExposureNotificationStatus = [Authorization, Enablement]
```

This fully captures the possible states and makes all others impossible. and makes the somewhat confusing situation mapped out for future readers.

and then in the view layer we'll have a pretty easy time pattern matching to show the correct message to the user.

Reply:
and the device status is what would be exposed to the view layer. not the authorization or enablement separately.

Reply:
or is it such that we cant be enabled if its unauthorized?

so we might want

```
type DeviceStatus = Enabled | Disabled | Unauthorized.
```

Reply:
^^ 💯